spring-kafka/src/test/java/org/springframework/kafka/core/KafkaTemplateTests.java

```diff
@@ -195,4 +195,14 @@ public void testTemplateDisambiguation() throws Exception {
 		pf.createProducer().close();
 	}
 
+	@Test(expected = IllegalStateException.class)
+	public void flushWithoutSend() throws Exception {
+		Map<String, Object> senderProps = KafkaTestUtils.producerProps(embeddedKafka);
+		DefaultKafkaProducerFactory<String, String> pf = new DefaultKafkaProducerFactory<String, String>(senderProps);
+		pf.setKeySerializer(new StringSerializer());
+		KafkaTemplate<String, String> template = new KafkaTemplate<>(pf);
+		template.setDefaultTopic(STRING_KEY_TOPIC);
+		template.flush();
+		throw new AssertionError("IllegalStateException must be thrown upon flushing");
```

Comment:
You got it!
Have a good night and come back to us with more feedback any time :smile: